### PR TITLE
Feature/split-debt-option

### LINKED ISSUE #94: Use Strategy pattern for determining whether a debt should be split between Users or each User should get a debt with the full amount.
i.e. SplitDebtStrategy and OneDebtForEachUserStrategy both implement IDebtStrategy and is used on Debt creation.
Let the user then select in the View whether to split the debt between the selected users or create one for each selected user.

### PR DESCRIPTION
**Pull request by Puh00**
* Implemented:
- IDebtSplitStrategy
     - EvenSplitStrategy
     - NoSplitStrategy
- Radiobuttons in AddDebtFragment to choose which strategy

Resolves #94

### CODE REVIEW

**Comment by Puh00** [comment]
Fails CI test but won't touch on tests since you guys have deleted them in your branch

File: app/src/main/java/com/goayo/debtify/model/EvenSplitStrategy.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @param borrowers The selected borrowers.
+     * @param owedTotal Amount that the borrowers are owed.
+     * @return A map with users and their respective owedTotal.
+     */
+    @Override
+    public Map<User, BigDecimal> splitDebt(Set<User> borrowers, BigDecimal owedTotal) {
+        // Splits the owedTotal by the number of borrowers
+        BigDecimal splitAmount = owedTotal
+                .divide(BigDecimal.valueOf(borrowers.size()), 10, RoundingMode.HALF_UP);
```
Comment (OlofSjogren) [inline app/src/main/java/com/goayo/debtify/model/EvenSplitStrategy.java:28]: Will this round up the second decimal or the first decimal or the whole number?
i.e. will: 
1.53 = 1.55? 
or 
1.53 = 1.50? 
or
1.53 = 2.0?

File: app/src/main/java/com/goayo/debtify/viewmodel/AddDebtViewModel.java [hunk 2]
```diff
@@ -74,18 +79,27 @@ public void setSelectedBorrowersData(Set<IUserData> borrowers) {
...
-                           String description) throws Exception {
+                           String description,
+                           boolean isNoSplit) throws Exception {
         // TODO refactor all strings to IUserData
+        IDebtSplitStrategy splitStrategy;
+        if (isNoSplit) {
+            splitStrategy = new NoSplitStrategy();
+        } else {
+            splitStrategy = new EvenSplitStrategy();
+        }
```
Comment (OlofSjogren) [inline app/src/main/java/com/goayo/debtify/viewmodel/AddDebtViewModel.java:97]: One thing we could do is create a ISplitStrategyFactory to remove the ViewModels dependency on the specific implementation of each splitStrategy and instead move the dependency to just the ISplitStrategy-interface.
ISplitStrategyFactory have methods "createNoSplitStrategy() : ISplitStrategy" and "createEvenSplitStrategy() : ISplitStrategy" which returns the specific ISplitStrategy implementation but it is wrapped in the generic ISplitStrategy interface.
This way ViewModels no longer create the object but just receive it wrapped from factory.

However we might be treading on "too much abstraction"-territory with this implementation. But it's worth a thought!

**Review by OlofSjogren — APPROVED** [review]
Great work!

**Pull request merged**